Clean up E0205 explanation (#68383)
rust-lang/rust: src/librustc_error_codes/error_codes/E0206.md

> The `Copy` trait was implemented on a type which isn't a struct or an enum.

Reviewer: ```suggestion
The `Copy` trait was implemented on a type which is neither a struct nor an enum.
```

Erroneous code example:

```compile_fail,E0206
type Foo = [u8; 256];
impl Copy for Foo { } // error!

#[derive(Copy, Clone)]
struct Bar;

impl Copy for &'static mut Bar { } // error!
```

> You can only implement `Copy` for a struct or enum. Both of the previous

Reviewer: ```suggestion
You can only implement `Copy` for a struct or an enum. Both of the previous
```
